### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Naledi-Dikgale**
**In this Pull Request we did the following**

- Set up React Testing Library.
- Wrote unit tests for the files operate.js and calculate.js using Jest.
- Created unit tests for all React components:
- Used Jest snapshots to test the components.
- Used React Testing Library to simulate user interaction.
- Ran full tests suite using Jest and ensured that all tests pass.

### CODE REVIEW

File: src/test.js/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+describe('testing operate.js function', () => {
+  test('expect tobe a string', () => {
+    expect(typeof operate(2, 2, '+')).toBe('string');
+  });
+
+  test('expect 3 * 3 to be 9', () => {
+    expect(operate(3, 3, 'x')).toBe('9');
+  });
+});
```
Comment (julie-ify) [inline src/test.js/operate.test.js]: - [ ] Kindly add more tests covering all operation signs in the logic file.

File: src/test.js/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  test('expect 0 and obj.next to be empty object', () => {
+    const obj = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    expect(calculate(obj, 'AC')).toEqual(obj);
+  });
+});
```
Comment (julie-ify) [inline src/test.js/calculate.test.js]: - [ ] Kindly write more tests covering more scenarios.

File: README.md [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 # 📖 Math Magician Project <a name="about-project"></a>
```
Comment (julie-ify) [inline README.md:28]: - [ ] You have added tests for only the logic files. Kindly add snapshot tests for all your components namely Calculator, Display, Home, Navigation, and Quote.

- [ ] Kindly add tests using the react testing library to simulate user interactions. You can see this tutorial: https://www.robinwieruch.de/react-testing-library/.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Naledi-Dikgale and @boniface193,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test.js/operate.test.js (hunk 1 above)
Comment (Abenezer-Tilahun) [reply src/test.js/operate.test.js]: Fixed.

File: src/test.js/calculate.test.js (hunk 2 above)
Comment (Abenezer-Tilahun) [reply src/test.js/calculate.test.js]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Naledi-Dikgale 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git -Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**